File: chart_scheduler/db/models/kr_tickers_model.py
from datetime import datetime
from typing import List
from sqlalchemy import Column, DECIMAL, String
from sqlalchemy.orm import relationship , backref
from sqlalchemy.dialects.mysql import TINYINT , DATE , SMALLINT
from chart_scheduler.db.models.kr_ticker_charts_model import KrTickerChartModel
from chart_scheduler.db.session import Base , meta_obj
from chart_scheduler.constants.schema_names import SCHEMA_NAMES

class KrTickerModel(Base) :
  __tablename__ = SCHEMA_NAMES.KR_TICKERS
  RELATIONSHIPS_TO_DICT = True
  __metaclass__=meta_obj

  symbol : str = Column(String(12) , nullable=False , index=True , primary_key=True)
  market : int = Column(TINYINT , nullable=True)
  index : int = Column(SMALLINT , nullable=True)
  name : str = Column(String(100) , nullable=False , index=True)
  bps : int = Column(DECIMAL(precision=11 , scale=2) , nullable=True , default=0)
  per : int = Column(DECIMAL(precision=11 , scale=2) , nullable=True , default=0)
  pbr : int = Column(DECIMAL(precision=11 , scale=2) , nullable=True , default=0)
  eps : int = Column(DECIMAL(precision=11 , scale=2) , nullable=True , default=0)
  div : int = Column(DECIMAL(precision=11 , scale=2) , nullable=True , default=0)
  dps : int = Column(DECIMAL(precision=11 , scale=2) , nullable=True , default=0)
  sector : str = Column(String(150) , nullable=True)
  industry : str = Column(String(150) , nullable=True)
  listing_date : datetime = Column(DATE , nullable=True)
  settle_month : str = Column(String(10) , nullable=True)
  representative : str = Column(String(100) , nullable=True)
  homepage : str = Column(String(200) , nullable=True)
  market_cap : int = Column(DECIMAL(precision=17 , scale=0) , nullable=True , default=None) 
  region : str = Column(String(10) , nullable=True)
  charts : List[KrTickerChartModel] = relationship(
    'KrTickerChartModel',
    back_populates='ticker',
    uselist=True
  )

File: gin/common/models/kr_tickers_model.go
package models

import (
	"time"

	"github.com/jjh930301/market/common/constants"
	"github.com/shopspring/decimal"
)

type KrTickerModel struct {
	Symbol         string                  `gorm:"column:symbol;type:varchar(12);not null;primary_key" json:"symbol,omitempty"` //index
	Merket         int                     `gorm:"column:market;type:tinyint" json:"market,omitempty"`
	Name           string                  `gorm:"column:name;type:varchar(100);not null" json:"name,omitempty"` //index
	Bps            decimal.Decimal         `gorm:"column:bps;type:decimal(11,2);default:0" json:"bps,omitempty"`
	Per            decimal.Decimal         `gorm:"column:per;type:decimal(11,2);default:0" json:"per,omitempty"`
	Pbr            decimal.Decimal         `gorm:"column:pbr;type:decimal(11,2);default:0" json:"pbr,omitempty"`
	Eps            decimal.Decimal         `gorm:"column:eps;type:decimal(11,2);default:0" json:"eps,omitempty"`
	Div            decimal.Decimal         `gorm:"column:div;type:decimal(11,2);default:0" json:"div,omitempty"`
	Dps            decimal.Decimal         `gorm:"column:dps;type:decimal(11,2);default:0" json:"dps,omitempty"`
	Sector         string                  `gorm:"column:sector;type:varchar(150)" json:"sector,omitempty"`
	Industry       string                  `gorm:"column:industry;type:varchar(150)" json:"industry,omitempty"`
	ListingDate    time.Time               `gorm:"column:listing_date;default:null" json:"listing_date,omitempty"`
	SettleMonth    string                  `gorm:"column:settle_month;type:varchar(10)" json:"settle_month,omitempty"`
	Representative string                  `gorm:"column:representative;type:varchar(100)" json:"representative,omitempty"`
	Homepage       string                  `gorm:"column:homepage;type:varchar(200)" json:"homepage,omitempty"`
	MarketCap      decimal.Decimal         `gorm:"column:market_cap;type:decimal(17,0);default:null" json:"market_cap,omitempty"`
	Region         string                  `gorm:"column:region;type:varchar(10)" json:"region,omitempty"`
	Chart          []KrTickerChartsModel   `gorm:"foreignKey:TickerSymbol" json:"chart,omitempty"`
	Comments       []KrTickerCommentModel  `gorm:"foreignKey:TickerSymbol" json:"comments,omitempty"`
	Interestes     []InterestedTickerModel `gorm:"foreignKey:TickerSymbol" json:"interests,omitempty"`
}

func (KrTickerModel) TableName() string {
	return constants.KrTickerTableName
}
